misc/mdns/src/service.rs

```diff
@@ -435,41 +462,22 @@ impl<'a> fmt::Debug for MdnsResponse<'a> {
 }
 
 /// A peer discovered by the service.
-pub struct MdnsPeer<'a> {
-    /// The original packet which will be used to determine the addresses.
-    packet: &'a Packet<'a>,
-    /// Cached value of `concat(base32(peer_id), service name)`.
-    record_value: String,
+pub struct MdnsPeer {
+    addrs: Vec<Multiaddr>,
     /// Id of the peer.
     peer_id: PeerId,
     /// TTL of the record in seconds.
     ttl: u32,
 }
 
-impl<'a> MdnsPeer<'a> {
-    /// Returns the id of the peer.
-    #[inline]
-    pub fn id(&self) -> &PeerId {
-        &self.peer_id
-    }
-
-    /// Returns the requested time-to-live for the record.
-    #[inline]
-    pub fn ttl(&self) -> Duration {
-        Duration::from_secs(u64::from(self.ttl))
-    }
-
-    /// Returns the list of addresses the peer says it is listening on.
-    ///
-    /// Filters out invalid addresses.
-    pub fn addresses<'b>(&'b self) -> impl Iterator<Item = Multiaddr> + 'b {
-        let my_peer_id = &self.peer_id;
-        let record_value = &self.record_value;
-        self.packet
+impl MdnsPeer {
+    /// Creates a new `MdnsPeer` based on the provided `Packet`.
+    pub fn new(packet: &Packet, record_value: String, my_peer_id: PeerId, ttl: u32) -> MdnsPeer {
```

Comment:
`MdnsPeer` is supposed to be a borrow for a value inside of a `Packet`, similar to a `MutexGuard` being a borrow of a `Mutex` for example.

Reply:
`Packet` is  not static, thus we need to do the parsing when constructing the `MdnsPeer` as the final `MdnsPeer` needs to be static.